plonky2/src/fri/recursive_verifier.rs

```diff
@@ -391,12 +433,20 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
     ) -> FriProofTarget<D> {
         let cap_height = params.config.cap_height;
         let num_queries = params.config.num_query_rounds;
-        let commit_phase_merkle_caps = (0..params.reduction_arity_bits.len())
+        let arities = if params.hiding {
+            let mut tmp = vec![1];
+            tmp.extend(&params.reduction_arity_bits);
```

Comment:
@LindaGuiga I see the thumbs up so I assume I was right, but I'm still curious. Shouldn't the reduction strategy take  this into account (as it is defined at the same time of the enablement of zk?)

Reply:
It's because de `degree_bits` (which are used to define other things, including, I think, the leaves of Merkle tries) are unchanged. So I changed the reduction strategy to include it indeed ( and removed all those parts in the code that still added the 1 to the arities)